integration-healthcheck/src/main/java/org/symphonyoss/integration/healthcheck/AsyncCompositeHealthIndicator.java

```diff
@@ -77,11 +72,15 @@
 
   private final LogMessageSource logMessageSource;
 
+  private final HealthCheckExecutorService service;
+
   @Autowired
-  public AsyncCompositeHealthIndicator(HealthAggregator aggregator, LogMessageSource logMessageSource) {
+  public AsyncCompositeHealthIndicator(HealthAggregator aggregator,
+      LogMessageSource logMessageSource, HealthCheckExecutorService service) {
     this.healthAggregator = aggregator;
     this.logMessageSource = logMessageSource;
-    this.indicators = new HashMap<>();
+    this.service = service;
+    this.indicators = new LinkedHashMap<>();
```

Comment:
Looks like we need a concurrent hashmap here as it is read by the async service.